src/pages/UpdateServers.js

```diff
@@ -155,20 +155,32 @@ class UpdateServers extends BaseUpdateWizardPage {
     let internalModelServers = this.state.internalModel.getIn(['internal', 'servers']).toJS();
     for (const server_id of serverIds.values()) {
       let server = internalModelServers.find(s => s.id == server_id);
+      if (!server) continue;
       try {
         const responseData = await fetchJson('/api/v2/monasca/server_status/' + server.hostname);
-        this.setState(prevState => {
-          let serverMonascaStatuses = prevState.serverMonascaStatuses;
-          serverMonascaStatuses[server_id] = translate('server.details.status.' + responseData.status);
-          return { 'serverMonascaStatuses' : serverMonascaStatuses };
-        });
+        this.setMonascaState(server_id, responseData.status);
       } catch(error) {
+        this.setMonascaState(server_id);
         console.log('error getting server status for:' + server.hostname + // eslint-disable-line no-console
           ' -- error is:' + error);
       }
     }
   }
 
+  /**
+   * Update the translated status of a server state
+   * @param {String} server_id The id of the server
+   * @param {String} state The current state of the server
```

Comment:
sorry to nitpick, but this variable should be called `status` rather than `state`, to avoid confusion with the `state` you are setting. I had to read this a couple of times to figure out that it was not related to `this.state`